onix-bot/onix.py

```diff
@@ -0,0 +1,150 @@
+import onixcheck
+import xml.etree.ElementTree as ET
+from lxml import etree
+
+# File 
+FILE = 'data/SampleONIX.xml'
+
+"""
+Step 1: Check if there are any errors in the XML file
+"""
+errors = onixcheck.validate(FILE)
+for error in errors:
+    print(error.short)
+
+# Exit the script in case there are errors
+if errors:
+    print("Kindly use a correct XML File to parse")
+    exit(0)
+
+"""
+Step 2: Parse the XML file to get all relevant features from the dataset
+"""
+tree = etree.parse(FILE)
+
+products = tree.xpath('/ONIXMessage/Product')
+
+# onix_record = [['Title','Publisher','City Of Publication','Country of Publication','isbn10', 'isbn13', 'Boocover Link', 'Language']]
+onix_records = [[]]
+
+for product in products:
+    product = etree.fromstring(etree.tostring(product))
```

Comment:
we can move the parsing logic for each book into its own function and this way we can test the function

e.g. `parse_product` can be a function which takes one `product` (xml) and returns a dictionary of a book (that Open Library client might use)

We can then write a test for `parse_product` and give it xml as a string and then verify that its output is the same as the dictionary we expect